File: client_mobile/app/src/main/java/com/example/tiktokapp/responseModel/Follow.java
package com.example.tiktokapp.responseModel;

import com.google.gson.annotations.SerializedName;

import java.sql.Timestamp;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class Follow extends AbstractModel{
    private final int follower;
    private final int followee;
    private int isFollowee;
    private int isFollow;
    private int isFriend;
    private int isMe;
    @SerializedName(value = "followerData", alternate = {"followeeData"})
    private User followData;

}

File: client/src/models/follow.ts
import { AbstractModel } from '.';
import { UserModel } from './user';

export interface FollowModel extends AbstractModel {
  follower: number;
  followerData: UserModel;
  followee: number;
  followeeData: UserModel;
  isFollow: number | boolean;
  isFriend: number | boolean;
  isFollowee: number | boolean;
}
